core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala

```diff
@@ -308,6 +303,23 @@ private[spark] class CoarseMesosSchedulerBackend(
     }
   }
 
+    // ToDo: Abstract out each condition and log them.
+  def isOfferSatisfiesRequirements(slaveId: String, mem: Double, cpusOffered: Int,
+                                   sc: SparkContext): Boolean = {
```

Comment:
also, the method signature is shared across both fine-grained and coarse-grained modes, so I would put this in the helper trait `MesosSchedulerUtils`.